examples/hello-world-python/README.md

```diff
@@ -0,0 +1,233 @@
+5. Build your Python files
+6. Test the Python library
+
+These steps wil be explained in detail below.
+
+
+### 1. Development Environment
+
+Let's first build our development container.
+
+```bash
+$ docker build -t helloworld .
+```
+
+We will want to bind the code directory (to write files and work interactively):
+
+```bash
+$ docker run -it --rm -v $PWD/:/code/ helloworld bash
+```
+
+If you don't want to use a container this is fine, just make sure you have basic
+dependencies on your host for compiling (build-essential) and a g++ compiler,
+along with shroud. At the time of this writing, installing from the master branch
+of the repository at [https://github.com/llnl/shroud](https://github.com/llnl/shroud)
+is your best bet, as the release on pypi has a bug with strings, and the extensions
+of yaml files.
+
+### 2. Write your C++ Scripts
+
+You can do this as your normally would without shroud. Shroud can handle namespaces
+and classes, and the example scripts show that well:
+
+
+ - [helloworld.cpp](helloworld.cpp)
+ - [helloworld.hpp](helloworld.hpp)
+  
+The reason they are in the same directory with the same name is that when I tried
+doing this differently, even when I added the cpp file to sources (which we will
+discuss later) it told me there was a missing symbol after build. This is
+the easiest way to ensure that doesn't happen.
+
+### 3. Write your shroud yaml file
+
+Shroud works by having you provide function and type signatures in a yaml file.
+You can also specify build options like "Please build Python bindings for this."
+Here is an example [helloworld.yaml](helloworld.yaml) that will generate for 
+our small "Person" class with two functions:
+
+```yaml
+library: helloworld
+
+# This must be relative to python directory targeted by shroud
+cxx_header: ../helloworld.hpp
+
+options:
+  wrap_fortran: False
+  wrap_c: False
+  wrap_python: True
+  debug: True
+
+declarations:
+- decl: namespace helloworld
+  declarations:
+  - decl: class Person
+    declarations:
+    - decl: int SayHello()
+    - decl: int NamedHello(std::string path)
+```
+This is a fairly small example, and you can look at the [shroud documentation](https://shroud.readthedocs.io/en/develop/tutorial.html)
+for further types. If you see something missing, please [open an issue](https://github.com/llnl/shroud) as it is
+still a work in progress.
+
+### 3. Compiling
+
+Ok, great! We have our source flies, and the shroud yaml. First, let me show you how to compile the files,
+separately from shroud. The commands are defined in the [Makefile](Makefile) and you will generate a binary,
+object file, and shared object file in the [bin](bin) directory (these are not added to the repository here
+as they are host dependent).
+
+```bash
+$ make
+g++ -o bin/helloworld -Iinclude helloworld.cpp -g -Wall 
+g++ -c -o bin/helloworld.o -fPIC -Iinclude helloworld.cpp -g -Wall 
+g++ -shared -o bin/helloworld.so bin/helloworld.o
+```
+```bash
```

Comment:
Remove this stray line.  It's starting a group that is not ended.

Reply:
Removed!